src/EnergyPlus/ChillerElectricEIR.cc

```diff
@@ -1866,18 +1867,24 @@ void ElectricEIRChillerSpecs::calculate(EnergyPlusData &state, Real64 &MyLoad, b
     // if the component control is SERIESACTIVE we set the component flow to inlet flow so that
     // flow resolver will not shut down the branch
     if (MyLoad >= 0 || !RunFlag) {
-        if (this->EquipFlowCtrl == DataBranchAirLoopPlant::ControlType::SeriesActive ||
-            state.dataPlnt->PlantLoop(this->CWPlantLoc.loopNum).LoopSide(this->CWPlantLoc.loopSideNum).FlowLock == DataPlant::FlowLock::Locked) {
-            this->EvapMassFlowRate = state.dataLoopNodes->Node(this->EvapInletNodeNum).MassFlowRate;
-        }
-        if (this->CondenserType == DataPlant::CondenserType::WaterCooled) {
-            if (DataPlant::CompData::getPlantComponent(state, this->CDPlantLoc).FlowCtrl == DataBranchAirLoopPlant::ControlType::SeriesActive) {
-                this->CondMassFlowRate = state.dataLoopNodes->Node(this->CondInletNodeNum).MassFlowRate;
-            }
-        }
-        if (this->CondenserType == DataPlant::CondenserType::EvapCooled) {
+        this->EvapMassFlowRate = 0.0;
+        PlantUtilities::SetComponentFlowRate(state, this->EvapMassFlowRate, this->EvapInletNodeNum, this->EvapOutletNodeNum, this->CWPlantLoc);
+
+        switch (this->CondenserType) {
+        case DataPlant::CondenserType::AirCooled: {
+            // nothing to do for air cooled
+        } break;
+        case DataPlant::CondenserType::WaterCooled: {
+            this->CondMassFlowRate = 0.0;
+            PlantUtilities::SetComponentFlowRate(state, this->CondMassFlowRate, this->CondInletNodeNum, this->CondOutletNodeNum, this->CDPlantLoc);
```

Comment:
This is inside `if (MyLoad >= 0 || !RunFlag)` so it's trying to tell the plant it has a zero flow request, but `PlantUtilities::SetComponentFlowRate` is returning a positive flow, because there's still flow on the chiller condenser inlet node at this point. But then later, plant flow resolution sets the inlet node mass flow rate to zero.

So this approach is causing `this->CondMassFlowRate` to be >0 at times when the chiller is off, because it never gets reset to zero, but the flow at the nodes is zero. I could just force `this->CondMassFlowRate` in the update function and maybe that's ok, but I'm just confused why `PlantUtilities::SetComponentFlowRate` is returning the inlet node flow rate when this is a simple branch with just the chiller condenser side on it.

Reply:
Tried adding this to the chiller `update` function when load is zero
```
           this->CondMassFlowRate = 0.0;
            PlantUtilities::SetComponentFlowRate(state, this->CondMassFlowRate, this->CondInletNodeNum, this->CondOutletNodeNum, this->CDPlantLoc);
```
And there are times when this comes back as 54, because flow is locked here.
https://github.com/NREL/EnergyPlus/blob/acda84e38db4e8aecbc178cb970055d3721ee321/src/EnergyPlus/PlantUtilities.cc#L270-L272

Reply:
It's EMS! Chiller is called with a load, `initialize` requests flow on both the evaporator and condenser sides. The evaporator mass flow rate comes back zero, because EMS supervisory control has shut off this chiller's evap branch (note that the evap side flowlock is unlocked), and the condenser flow comes back at max flow as requested.

In `calculate` the zero evap mass flow rate results in `MyLoad=0` and `return` without resetting the condender flow rate to zero. But even setting that to zero and calling `setcomponentflowrate` doesn't work, because the condenser side is already flowlocked at the earlier flow request. 

In `update` if `MyLoad ==0` it forces zero flow on the condenser nodes without using `setcomponentflowrate` (which wouldn't work anyway because the condenser side is locked at max flow.

That's enough twists and turns for today.

Reply:
So then the condenser plant FlowLock needs to be conditional on the FlowLock for the plant it serves?